test/library/packages/UnitTest/Launcher.chpl

```diff
@@ -21,86 +21,153 @@ module Launcher {
         files: [1..0] string;
     var hadInvalidFile = false;
     var programName = args[0];
-
-    sys_getenv("CHPL_COMM",comm_c);
-    comm = comm_c: string;
-    
-    for a in args[1..] {
-      if a == "-h" || a == "--help" {
-        writeln("Usage: ", programName," <options> filename [filenames] directoryname [directorynames]");
-        printUsage();
-        exit(1); // returning 1 from main is also an option
-      }
-      else {
-        try! {
-          if isFile(a) {
-            files.push_back(a);
-          }
-          else if isDir(a) {
-            dirs.push_back(a);
+    try! {
+      var checkChpl = spawn(["which","chpl"],stdout = PIPE);
+      checkChpl.wait();
+      var line: string;
+      if checkChpl.stdout.readline(line) {
+        sys_getenv("CHPL_COMM", comm_c);
+        comm = comm_c: string;
+        
+        for a in args[1..] {
+          if a == "-h" || a == "--help" {
+            writeln("Usage: ", programName, " <options> filename [filenames] directoryname [directorynames]");
+            printUsage();
+            exit(1); // returning 1 from main is also an option
```

Comment:
Doesn't exiting with a non-zero exit code indicate that something is wrong?  Printing the help output when `-h` or `--help` is provided seems like expected behavior, so I would expect the program to exit with `0`.

Reply:
Okay Ma'am.